controller/internal/supervisor/iptablesctrl/iptables.go

```diff
@@ -817,3 +857,102 @@ func flushUDPConntrack(networks []string) {
+//   - ExcludedNetworks that are always ignored (by default empty)
+func (i *Instance) SetTargetNetworks(c *runtime.Configuration) error {
+
+	if err := i.iptv4.SetTargetNetworks(c); err != nil {
+		return err
+	}
+
+	if err := i.iptv6.SetTargetNetworks(c); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Run starts the iptables controller
+func (i *Instance) Run(ctx context.Context) error {
+
+	if err := i.iptv4.Run(ctx); err != nil {
+		return err
+	}
+
+	if err := i.iptv6.Run(ctx); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// ConfigureRules implments the ConfigureRules interface. It will create the
+// port sets and then it will call install rules to create all the ACLs for
+// the given chains. PortSets are only created here. Updates will use the
+// exact same logic.
+func (i *Instance) ConfigureRules(version int, contextID string, pu *policy.PUInfo) error {
+	if err := i.iptv4.ConfigureRules(version, contextID, pu); err != nil {
+		return err
+	}
+
+	if err := i.iptv6.ConfigureRules(version, contextID, pu); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// DeleteRules implements the DeleteRules interface. This is responsible
+// for cleaning all ACLs and associated chains, as well as ll the sets
+// that we have created. Note, that this only clears up the state
+// for a given processing unit.
+func (i *Instance) DeleteRules(version int, contextID string, tcpPorts, udpPorts string, mark string, username string, proxyPort string, puType common.PUType) error {
+
+	if err := i.iptv4.DeleteRules(version, contextID, tcpPorts, udpPorts, mark, username, proxyPort, puType); err != nil {
+		zap.L().Warn("Delete rules for iptables v4 returned error")
+	}
+
+	if err := i.iptv6.DeleteRules(version, contextID, tcpPorts, udpPorts, mark, username, proxyPort, puType); err != nil {
+		zap.L().Warn("Delete rules for iptables v6 returned error")
+	}
+
+	return nil
+}
+
+// UpdateRules implements the update part of the interface. Update will call
+// installrules to install the new rules and then it will delete the old rules.
+// For installations that do not have latests iptables-restore we time
+// the operations so that the switch is almost atomic, by creating the new rules
+// first. For latest kernel versions iptables-restorce will update all the rules
+// in one shot.
+func (i *Instance) UpdateRules(version int, contextID string, containerInfo *policy.PUInfo, oldContainerInfo *policy.PUInfo) error {
+
+	if err := i.iptv4.UpdateRules(version, contextID, containerInfo, oldContainerInfo); err != nil {
+		return err
+	}
+
+	if err := i.iptv6.UpdateRules(version, contextID, containerInfo, oldContainerInfo); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// CleanUp requires the implementor to clean up all ACLs and destroy all
+// the IP sets.
+func (i *Instance) CleanUp() error {
+
+	if err := i.iptv4.CleanUp(); err != nil {
+		return err
+	}
+
+	if err := i.iptv6.CleanUp(); err != nil {
+		return err
+	}
+
```

Comment:
Cleanups should move on to ipv6 even if there is an error in ipv4. We need to try to clean up as much as we can.